Jenkinsfile

```diff
@@ -125,26 +125,52 @@ pipeline {
-                    sh "echo starting e2e tests" 
-                    sh "make test-e2e"
-                    sh "make test-esx"
-                    sh "make test-vm"
+                    steps {
+		        script {
+			    try {
+                                sh "echo Test VDVS On 6.5 setup"
+                                sh "echo ESX=$ESX; echo VM1=$VM1; echo VM2=$VM2; echo VM3=$VM3; echo PKG_VERSION"
+                                sh "echo starting e2e tests"
+                                sh "make test-e2e"
+                                sh "make test-esx"
+                                sh "make test-vm"
+			    }
+			    catch (ex) {
+                                sh "ssh ${env.GOVC_USERNAME}@$VM1 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "ssh ${env.GOVC_USERNAME}@$VM2 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "ssh ${env.GOVC_USERNAME}@$VM3 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "make clean-all"
+			        throw ex
+			    }
+			}
                     }
                 }
+
                 stage('Run tests on ESX 6.0') {
                     agent {
-                    label env.VDVS_60_NODE_ID
+                        label env.VDVS_60_NODE_ID
                     }   
-                steps {
-                    sh "echo Test VDVS On 6.0 setup"
-                    sh "echo ESX=$ESX; echo VM1=$VM1; echo VM2=$VM2; echo VM3=$VM3; echo PKG_VERSION"
-                    sh "echo starting e2e tests"
-                    sh "make test-e2e"
-                    sh "make test-esx"
-                    sh "make test-vm"
+                    steps {
+		        script {
+                            def stopContainers = "docker stop \$(docker ps -a -q) 2> /dev/null || true"
+                            def removeContainers = "docker rm \$(docker ps -a -q) 2> /dev/null || true"
+                            def removeVolumes =  "docker volume rm \$(docker volume ls -q -f dangling=true) 2> /dev/null || true"
+			    try {
+                                sh "echo Test VDVS On 6.0 setup"
+                                sh "echo ESX=$ESX; echo VM1=$VM1; echo VM2=$VM2; echo VM3=$VM3; echo PKG_VERSION"
+                                sh "echo starting e2e tests"
+                                sh "make test-e2e"
+                                sh "make test-esx"
+                                sh "make test-vm"
+			    }
+			    catch (ex) {
+                                sh "ssh ${env.GOVC_USERNAME}@$VM1 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "ssh ${env.GOVC_USERNAME}@$VM2 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "ssh ${env.GOVC_USERNAME}@$VM3 ${stopContainer}; ${removeContainer}; ${removeVolume}"
+                                sh "make clean-all"
+			        throw ex
```

Comment:
nit: there seems to a problem with indentations i.e. there's a mix of tabs and spaces in this block.